entity-server-api/src/main/java/org/terracotta/entity/EntityServerService.java

```diff
@@ -64,12 +64,51 @@
   PassiveServerEntity<M, R> createPassiveEntity(ServiceRegistry registry, byte[] configuration);
   
   /**
+   * Reconfigure an existing entity during server execution.  Reconfigure call in the
+   * platform's entity API allows an entity to reconfigure in place, its instance in the 
+   * system.  Prior to the call of this method, the entities execution queue is flushed and 
+   * any subsequent messages are blocked until this call completes.  This call occurs only 
+   * under the MANAGEMENT_KEY concurrency key.  Implementations are allowed to either reconfigure
+   * in place and return the same instance of the entity or a new entity may be built if desired.
+   * 
+   * @param <AP> the entity type being reconfigured
+   * @param registry the service registry for the entity
+   * @param oldEntity the instance of the entity implementation being reconfigured
+   * @param configuration the new configuration of the entity
+   * @return An entity instance influenced by the new configuration
+   */
+  default <AP extends CommonServerEntity<M, R>> AP reconfigureEntity(ServiceRegistry registry, AP oldEntity, byte[] configuration) {
```

Comment:
This does not involve a major state transition of the server.  Connected clients and the invoke queue can be assured to be consistent during the reconfigure call.  With failover, we are dealing with a major state transition of the server where the availability of clients can change during transition.  

Having said that, I am not 100% comfortable with this addition to the API.  I think an option may be to remove reconfigure and force removal and re-creation of entities to produce a similar result.  Coupled with some of the discussions around higher level entity management, this may be a better solution.